### PR TITLE
Add history for 5.4.0-m4

### PR DESCRIPTION
**Pull request by jburel**
# What this PR does

Add entry for m4

# Testing this PR
Nothing to test

### CODE REVIEW

File: history.txt [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+5.4.0-m4 (August 2017)
+----------------------
+
+Developer preview release. Intended *only as a developer preview* for updating
+code before the full public release of 5.4.0. **Use at your own risk.**
+
+Changes include:
+
+- added method to JavaGateway to manipulate admin privileges
+- fixed graph policy rule to avoid stack overflow wheb moving large plates at once
```
Comment (mtbc) [inline history.txt]: "when"

File: history.txt [hunk 2]
```diff
@@ -33,6 +33,11 @@ Changes include:
 - created minimal nginx configuration file that can be included in a fixed file
   to allow custom nginx options to be defined once (e.g. SSL options)
 - closed services when 404 occurs
+- dropped support for CLI command ``admin ports``
+- added support to the Rendering engine to update a series of settings in one call
+- added infrastructure to load external CLI plugins
+- allowed the maven repository to be overriden
```
Comment (mtbc) [inline history.txt]: overridden

File: history.txt [hunk 3]
```diff
@@ -1,6 +1,49 @@
...
+
+Developer preview release. Intended *only as a developer preview* for updating
+code before the full public release of 5.4.0. **Use at your own risk.**
+
+Changes include:
+
+- added method to JavaGateway to manipulate admin privileges
+- fixed graph policy rule to avoid stack overflow when moving large plates at once
+- fixed links to point to new website
+- fixed color convertion to rgba
```
Comment (hflynn) [inline history.txt]: typo:
convertion -> conversion

Comment (hflynn) [reply history.txt]: and don't we usually capitalize RGBA?

File: history.txt [hunk 4]
```diff
@@ -1,6 +1,49 @@
...
+- fixed export as TIFF from webgateway
+- improved creating of rendering settings for image without any stats
```
Comment (hflynn) [inline history.txt]: creating -> creation

File: history.txt [hunk 5]
```diff
@@ -1,6 +1,49 @@
...
+  e.g. 32bit images
+- fixed issue with jsonp decorator
```
Comment (hflynn) [inline history.txt]: JSONP

File: history.txt [hunk 6]
```diff
@@ -1,6 +1,49 @@
...
+- fixed upload of script for admin with restricted privileges
```
Comment (hflynn) [inline history.txt]: script -> scripts

File: history.txt [hunk 7]
```diff
@@ -1,6 +1,49 @@
...
+- fixed import bug in OMERO.insight
+- improved copy/paste of rendering setttings workflow in OMERO.web
+- fixed projection of images if the range of timepoint specified is not
```
Comment (hflynn) [inline history.txt]: timepoint -> timepoints

File: history.txt [hunk 8]
```diff
@@ -1,6 +1,49 @@
...
+  the full range
+- added configuration to limit query that public user can do in OMERO.web
```
Comment (hflynn) [inline history.txt]: query -> queries
user -> users

File: history.txt [hunk 9]
```diff
@@ -1,6 +1,49 @@
...
+- improved display of wells in OMERO.web
+- allowed admin with restricted privileges to create P/D for others in
```
Comment (hflynn) [inline history.txt]: I assume P/D is projects and datasets but it took me a minute to work that out so best to write it out properly
Also:
admin -> administrators

File: history.txt [hunk 10]
```diff
@@ -1,6 +1,49 @@
...
+  OMERO.web
+- allowed the creation of admin with restricted privileges with password
```
Comment (hflynn) [inline history.txt]: what's the 'with password' bit here for? It reads strangely.

File: history.txt [hunk 11]
```diff
@@ -1,6 +1,49 @@
...
+- added support to the creation of admin with restricted privileges in
```
Comment (hflynn) [inline history.txt]: to -> for?
admin -> administrators (and above)

File: history.txt [hunk 12]
```diff
@@ -1,6 +1,49 @@
...
+  OMERO.web
+- reverted workaround for IcePy communicator leak
+- created minimal nginx configuration file that can be included in a fixed file
+  to allow custom nginx options to be defined once (e.g. SSL options)
```
Comment (hflynn) [inline history.txt]: once -> only once

File: history.txt [hunk 13]
```diff
@@ -1,6 +1,49 @@
...
+- closed services when 404 occurs
+- dropped support for CLI command ``admin ports``
+- added support to the Rendering engine to update a series of settings in one call
+- added infrastructure to load external CLI plugins
+- allowed the maven repository to be overridden
+- rolled back the display of tables in OMERO.web viewer
+
+
```
Comment (hflynn) [inline history.txt]: Only need one line of whitespace either side of the BF upgrade line

File: history.txt (hunk 10 above)
Comment (mtbc) [reply history.txt]: You could already create restricted admins, this adds a new method for setting a password for them when creating them.

Comment (hflynn) [reply history.txt]: In that case:
"added support for setting a password for restricted administrators when creating them"

File: history.txt [hunk 14]
```diff
@@ -1,6 +1,52 @@
...
+- fixed graph policy rule to avoid stack overflow when moving large plates at once
+- fixed links to point to new website
+- fixed color conversion to RGBA
+- fixed export as TIFF from webgateway
+- improved creation of rendering settings for image without any stats
+  e.g. 32bit images
+- fixed issue with JSONP decorator
+- fixed upload of scripts for administrators with restricted privileges
+- fixed import bug in OMERO.insight
+- improved copy/paste of rendering setttings workflow in OMERO.web
```
Comment (hflynn) [inline history.txt]: setttings typo - one too many 't's

**Comment by jburel** [comment]
@hflynn: everything should now be fixed

**Comment by jburel** [comment]
@sbesson @hflynn entry for 5.3.4 cherry-picked and added to this PR

**Comment by jburel** [comment]
@pwalczysko I have added missing ``chown`` feature

**Comment by pwalczysko** [comment]
@jburel : looks good, thank you

**Comment by hflynn** [comment]
@jburel are you going to merge this and then open a new PR to combine the milestones into one 5.4.0 history? Are we doing an m4 internal release?

File: history.txt [hunk 15]
```diff
@@ -51,9 +51,15 @@ Changes include:
 - improved handling of images without stats to determine rendering settings
 - added options to export images in plate as OME-TIFF
 - added ability to transfer all the data of a given user using CLI
+- added support for PyTables version 3.4+
+- deprecated Path Object in OMERO model
+- updated url to point to new website, docs and downloads
```
Comment (hflynn) [inline history.txt]: url -> URL
Otherwise 👍

**Comment by jburel** [comment]
Merging so we can prepare m4 tomorrow

**Pull request merged**